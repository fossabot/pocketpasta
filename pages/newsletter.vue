<template>
  <div class="container">
    <!-- Begin MailChimp Signup Form -->
    <div id="mc_embed_signup">
      <form
        id="mc-embedded-subscribe-form"
        action="https://pocketpasta.us16.list-manage.com/subscribe/post?u=7f69c361c7f5d97725d2e339b&amp;id=c821016b08"
        method="post"
        name="mc-embedded-subscribe-form"
        class="validate"
        target="_blank"
        novalidate
      >
        <div id="mc_embed_signup_scroll">
          <h1>Subscribe to our mailing list</h1>
          <div class="indicates-required">
            <span class="asterisk">*</span> indicates required
          </div>
          <div class="mc-field-group">
            <label for="mce-EMAIL">
              Email Address
              <span class="asterisk">*</span>
            </label>
            <input
              id="mce-EMAIL"
              type="email"
              value
              name="EMAIL"
              class="required email"
            />
          </div>
          <div class="mc-field-group">
            <label for="mce-FNAME">First Name</label>
            <input id="mce-FNAME" type="text" value name="FNAME" class />
          </div>
          <div class="mc-field-group">
            <label for="mce-LNAME">Last Name</label>
            <input id="mce-LNAME" type="text" value name="LNAME" class />
          </div>
          <div id="mce-responses" class="clear">
            <div
              id="mce-error-response"
              class="response"
              style="display:none"
            />
            <div
              id="mce-success-response"
              class="response"
              style="display:none"
            />
          </div>
          <!-- real people should not fill this in and expect good things
        - do not remove this or risk form bot signups-->
          <div style="position: absolute; left: -5000px;" aria-hidden="true">
            <input
              type="text"
              name="b_7f69c361c7f5d97725d2e339b_c821016b08"
              tabindex="-1"
              value
            />
          </div>
          <div class="clear">
            <input
              id="mc-embedded-subscribe"
              type="submit"
              value="Subscribe"
              name="subscribe"
              class="button"
            />
          </div>
        </div>
      </form>
    </div>
    <!--End mc_embed_signup-->
  </div>
</template>

<script>
export default {
  head() {
    return {
      link: [
        {
          rel: 'canonical',
          href: 'https://pocketpasta.com/newsletter',
        },
      ],
    };
  },
  methods: {
    subscribe() {
      this.$ga.event('Newsletter', 'Click', 'Subscribe');
    },
  },
};
</script>

<style>
@import url('//cdn-images.mailchimp.com/embedcode/classic-10_7.css');
</style>
