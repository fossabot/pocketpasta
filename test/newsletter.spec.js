import { shallowMount } from '@vue/test-utils';
import newsletter from '@/pages/newsletter.vue';

const factory = () =>
  shallowMount(newsletter, {
    mocks: {
      $ga: { event: () => true },
    },
  });

describe('newsletter', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('head', () => {
    expect(newsletter.head()).toBeTruthy();
  });

  test('subscribe event', () => {
    const wrapper = factory();
    expect(wrapper.vm.subscribe()).toBe();
  });
});
