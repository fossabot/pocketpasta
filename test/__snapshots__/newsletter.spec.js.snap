// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newsletter renders properly 1`] = `
"<div class=\\"container\\"><div id=\\"mc_embed_signup\\"><form id=\\"mc-embedded-subscribe-form\\" action=\\"https://pocketpasta.us16.list-manage.com/subscribe/post?u=7f69c361c7f5d97725d2e339b&amp;id=c821016b08\\" method=\\"post\\" name=\\"mc-embedded-subscribe-form\\" target=\\"_blank\\" novalidate=\\"novalidate\\" class=\\"validate\\"><div id=\\"mc_embed_signup_scroll\\"><h1>newsletter.heading</h1> <div class=\\"indicates-required\\"><span class=\\"asterisk\\">*</span> newsletter.required
        </div> <div class=\\"mc-field-group\\"><label for=\\"mce-EMAIL\\">
            newsletter.email
            <span class=\\"asterisk\\">*</span></label> <input id=\\"mce-EMAIL\\" type=\\"email\\" name=\\"EMAIL\\" class=\\"required email\\"></div> <div class=\\"mc-field-group\\"><label for=\\"mce-FNAME\\">newsletter.firstname</label> <input id=\\"mce-FNAME\\" type=\\"text\\" name=\\"FNAME\\"></div> <div class=\\"mc-field-group\\"><label for=\\"mce-LNAME\\">newsletter.lastname</label> <input id=\\"mce-LNAME\\" type=\\"text\\" name=\\"LNAME\\"></div> <div id=\\"mce-responses\\" class=\\"clear\\"><div id=\\"mce-error-response\\" class=\\"response\\" style=\\"display: none;\\"></div> <div id=\\"mce-success-response\\" class=\\"response\\" style=\\"display: none;\\"></div></div> <div aria-hidden=\\"true\\" style=\\"position: absolute; left: -5000px;\\"><input type=\\"text\\" name=\\"b_7f69c361c7f5d97725d2e339b_c821016b08\\" tabindex=\\"-1\\" value=\\"\\"></div> <div class=\\"clear\\"><input id=\\"mc-embedded-subscribe\\" type=\\"submit\\" name=\\"subscribe\\" class=\\"button\\" value=\\"newsletter.submit\\"></div></div></form></div></div>"
`;
